### PR TITLE
Bump go-multihash, adjust test for supported version

### PR DESCRIPTION
**Pull request by ribasushi**
Snake through a fix for a subtle test failure. Incorporates https://github.com/filecoin-project/specs-actors/pull/1463#issue-680772451

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -48,7 +48,7 @@ require (
 	github.com/filecoin-project/specs-actors/v2 v2.3.5
 	github.com/filecoin-project/specs-actors/v3 v3.1.1
 	github.com/filecoin-project/specs-actors/v4 v4.0.1
-	github.com/filecoin-project/specs-actors/v5 v5.0.1
+	github.com/filecoin-project/specs-actors/v5 v5.0.2-0.20210630111506-5d9e79a8e57e
```
Comment (magik6k) [inline go.mod:51]: Can we get this into master / tagged?

Comment (ribasushi) [reply go.mod:51]: @magik6k that'll be a *very* long time...
https://github.com/filecoin-project/specs-actors/pull/1463#issuecomment-871475927

Comment (ribasushi) [reply go.mod]: @magik6k this now tracks current specs-actors master, I did not realize we can make the change there non-contentious.

Comment (ZenGround0) [reply go.mod]: We can use https://github.com/filecoin-project/specs-actors/releases/tag/v5.0.3 now

Comment (ribasushi) [reply go.mod]: Thanks @ZenGround0, adjusted

**Review by arajasek — APPROVED** [review]
Nice

**Pull request merged**